### ISSUE TITLE
Cannot assign a TCEMainForm to a TCEPersistentMainShortcuts due to docking.xml

### ISSUE THREAD

**Issue by shishini** [BBasile/Coedit#250]
i am running windows 7 service pack 1
and coedit 3.6.3

suddenly and without making changes to my system i get this error in a popup when i try to open coedit 

```
Cannot assing a TCEMainForm to a TCEPersistentMainShortcuts
Press Ok to ignore and risk data corruption
Press Abort to kill the program
```

i tried to uninstall and reinstall, but this doesnt seem to change anything

**Comment by BBasile** [comment]
Try this: delete `C:\Users\<your account>\AppData\Roaming\Coedit\mainshortcuts.txt` and restart.

**Comment by shishini** [comment]
hi

i did try deleting this one file 
but nothing changed

i delete the whole folder C:\Users\<your account>\AppData\Roaming\Coedit

not it works 
thanks

i am only using it for learning, so i was not worried about loosing any data

**Comment by BBasile** [comment]
You should have deleted them one by one. Now i still don't know what was happening and this problem looked serious. If you still have them in the trashcan id'l like to know. So restore the files and then one after the other delete it, start CE. When there's not the error message anymore tell me which is the file that was deleted for this step.

**Comment by shishini** [comment]
luckily i had two accounts on the computer with the same issue (one admin account and one regular account)

i did as you asked and the one file that seem to cause the issue is the one called
docking.xml

**Comment by BBasile** [comment]
Thanks.

**Comment by shishini** [comment]
Hi 

i think i can add more details on why this happens, 
i didnt make the link at first, but the file name helped me a little 

i use a laptop, at work, that i connect to a large screen, and in this setup the external large screen
act as my main screen, so this is where all windows open by defaults and where i see my task bar

the external screen is much larger that my laptop screen

the error happens when i switch my main screen, to the laptop (this happens when i unplug the screen)

now that i returned to work, i dont have this error any more using the same doking.xml and my admin account 

and now on my regular account, where i deleted doking.xml, i see a new side effect error 
see screenshot attached 
![image](https://user-images.githubusercontent.com/2177439/36381808-c20927a0-1554-11e8-9983-86036107a612.png)

so in summary, when i move my main screen from large to small, i get the TCEMainForm error
when i move from small to large i get he undesired side effect as per the screen shot

**Comment by BBasile** [comment]
CTRL + drag to change the top splitter position (just above the caption "Source Editor").

**Comment by shishini** [comment]
i fixed it by deleting the docking.xml file 
i didn guess that i need to press ctrl+drag , and just dragging didnt work 
anyway, i think the docking.xml file should not be cached

**Comment by BBasile** [comment]
For your information docking has been the top one source of problem since the beginning. It's a visual component that i didn't write and is in beta phase since forever. Unfortunately i don't wish to make GUI component programming anymore, that's why for example the option editor use a kind of table based on RTTI. There's very few custom GUI stuff in Coedit.

**Comment by BBasile** [comment]
This was more a support ticket. Feel free to open something else when you can describe how to reproduce.
